utility.hpp.in

```diff
@@ -13,6 +13,8 @@
  */
 namespace utility{
 
+template <typename T> class Id;
```

Comment:
Minor quible: I'd prefer this forward declaration be placed in the section labeled `/* forward declaration of classes */` in comments